.github/actions/change-detection/action.yml

```diff
@@ -0,0 +1,141 @@
+name: Detect changed files
+description: Detects whether any matching files have changed in the current PR
+inputs:
+  self:
+    description: The path to the calling workflow (e.g. .github/workflows/backend-unit-tests.yml). It is regarded as any category.
+    required: true
+outputs:
+  self:
+    description: Whether the calling workflow has changed in any way
+    value: ${{ steps.changed-files.outputs.self_any_changed || 'true' }}
+  backend:
+    description: Whether the backend or self have changed in any way
+    value: ${{ steps.changed-files.outputs.backend_any_changed || 'true' }}
+  frontend:
+    description: Whether the frontend or self has changed in any way
+    value: ${{ steps.changed-files.outputs.frontend_any_changed || 'true' }}
+  e2e:
+    description: Whether the e2e tests or self have changed in any way
+    value: ${{ steps.changed-files.outputs.e2e_any_changed == 'true' ||
+      steps.changed-files.outputs.backend_any_changed == 'true' ||
+      steps.changed-files.outputs.frontend_any_changed == 'true' || 'true' }}
+  dev-tooling:
+    description: Whether the dev tooling or self have changed in any way
+    value: ${{ steps.changed-files.outputs.dev_tooling_any_changed || 'true' }}
+  docs:
+    description: Whether the docs or self have changed in any way
+    value: ${{ steps.changed-files.outputs.docs_any_changed || 'true' }}
+runs:
+  using: composite
+  steps:
+    # Assumption: We've done a checkout with the actions/checkout action.
+    # It must persist credentials to allow the changed-files action to get more history.
+    - name: Detect changes
+      id: changed-files
+      if: github.event_name == 'pull_request'
+      uses: tj-actions/changed-files@ed68ef82c095e0d48ec87eccea555d944a631a4c # v46
+      with:
+        files_yaml: |
+          self:
+            - '.github/actions/change-detection/**'
+            - '${{ inputs.self }}'
+          backend:
+            - '!*.md'
+            - '!docs/**'
+            - '!.github/**'
+            - '.github/actions/setup-enterprise/**'
+            - '.github/actions/checkout/**'
+            - '**/go.mod'
+            - '**/go.sum'
+            - '**.go'
+            - 'pkg/**'
+            - '!pkg/**.md'
+            - 'apps/**'
+            - '!apps/**.md'
+            - 'build.sh'
+            - '.github/actions/change-detection/**'
+            - '**.cue'
+            - 'devenv/docker/blocks/*_tests/**'
+            - 'kindsv2/**'
+            - '${{ inputs.self }}'
+          frontend:
+            - '.github/actions/setup-enterprise/**'
+            - '.github/actions/checkout/**'
+            - 'public/**'
+            - '**.js'
+            - '**.jsx'
+            - '**.ts'
+            - '**.tsx'
+            - '**.css'
+            - '**.mjs'
+            - 'yarn.lock'
+            - 'package.json'
+            - '!**.md'
```

Comment:
I believe we still want to run `prettier:check` when changes are made to `*.md` files.

I think that's why https://github.com/grafana/grafana/pull/107587 didn't fail any check (only Markdown files were edited, so the CI step was skipped).

Any changes [_after_ that](https://github.com/grafana/grafana/actions/runs/16176126405/job/45661591965) that do include source files will start failing because the Markdown file isn't linted.